rust/src/ikev1/ikev1.rs

```diff
@@ -0,0 +1,606 @@
+                                    if direction == STREAM_TOSERVER { &mut self.client_nonce } else { &mut self.server_nonce },
+                                    if direction == STREAM_TOSERVER { &mut self.client_transforms } else { &mut self.server_transforms },
+                                    if direction == STREAM_TOSERVER { &mut self.client_vendor_ids } else { &mut self.server_vendor_ids },
+                                    &mut payload_types
+                                ) {
+                                    SCLogDebug!("Error while parsing IKEV1 payloads");
+                                    return AppLayerResult::err();
+                                }
+
+                                cur_payload_type = isakmp_payload.payload_header.next_payload;
+                            }
+                        },
+                        Err(nom::Err::Incomplete(_)) => {
+                            SCLogDebug!("Insufficient data while parsing IKEV1");
+                            return AppLayerResult::err();
+                        }
+                        Err(_) => {
+                            return AppLayerResult::err();
+                        }
+                    }
+                }
+
+                let mut tx = self.new_tx();
+                tx.spi_initiator = Some(isakmp_header.init_spi);
+                tx.spi_responder = Some(isakmp_header.resp_spi);
+                tx.maj_ver = Some(isakmp_header.maj_ver);
+                tx.min_ver = Some(isakmp_header.min_ver);
+                tx.exchange_type = Some(isakmp_header.exch_type);
+                tx.payload_types = Some(payload_types);
+                tx.encrypted_payloads = encrypted_payloads;
+                self.transactions.push(tx);
+                self.set_progress();
+
+                return AppLayerResult::ok(); // todo either remove outer loop or check header length-field if we have completely read everything
+            }
+            Err(nom::Err::Incomplete(_)) => {
+                SCLogDebug!("Insufficient data while parsing IKEV1");
+                return AppLayerResult::err();
+            }
+            Err(_) => {
+                return AppLayerResult::err();
+            }
+        }
+    }
+
+    fn tx_iterator(
+        &mut self,
+        min_tx_id: u64,
+        state: &mut u64,
+    ) -> Option<(&IKEV1Transaction, u64, bool)> {
+        let mut index = *state as usize;
+        let len = self.transactions.len();
+
+        while index < len {
+            let tx = &self.transactions[index];
+            if tx.tx_id < min_tx_id + 1 {
+                index += 1;
+                continue;
+            }
+            *state = index as u64;
+
+            // return Some((tx, tx.tx_id - 1, (len - index) > 1)); <- original
+
+            // todo: ask oisf next 4 lines! only return last transaction in iterator? seems to return correct number of alerts
```

Comment:
forget what i've written here, with changing logic to use the "immutable" transaction holding the values for alerting instead of the state the counts look much better now without this hack.